lib/bosh/BoshDirectorClient.js

```diff
@@ -169,6 +183,16 @@ class BoshDirectorClient extends HttpClient {
     return this.request(requestDetails, expectedStatusCode);
   }
 
+  checkAndUpdateCache(deploymentName, config) {
```

Comment:
No need to define this new method. Rather put this implementation into the method [here](https://github.com/SAP/service-fabrik-broker/blob/master/lib/bosh/BoshDirectorClient.js#L141-L153). Current implementation in getDirectorConfig reloads entire cache when there is a miss. Rather it must only update the cache entry for the deployment name which is not found.